### PR TITLE
Ensuring we use at least boost 1.65, build static on Xenial, and adapt CircleCI accordingly.

### PR DESCRIPTION
**Pull request by christianparpart**
This PR is to make sure we use at least Boost 1.65 on CircleCI and 1.67 @ AppVeyor. - before it was 1.62 and 1.61.

**Motivation:** boost 1.65 is the first to also provide `boost::get` for rvalue-references.

For AppVeyor, I've directly upgraded to Boost 1.67 (I already had it prebuilt and just needed to package & upload).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -315,7 +315,10 @@ jobs:
 
   docs:
     docker:
-      - image: buildpack-deps:artful
+      - image: buildpack-deps:bionic
+    environment:
+      DEBIAN_FRONTEND: noninteractive
```
Comment (ekpyron) [inline .circleci/config.yml:320]: Can this be moved up to ``defaults``? Not sure whether there can be an ``environment`` field in ``defaults``, but in general it makes sense to have this defined in all runs.

File: .circleci/config.yml [hunk 2]
```diff
@@ -315,7 +315,10 @@ jobs:
...
+      TZ: UTC
```
Comment (ekpyron) [inline .circleci/config.yml]: Do we still need this given the last line?

File: cmake/EthDependencies.cmake [hunk 3]
```diff
@@ -5,6 +5,9 @@ function(eth_show_dependency DEP NAME)
 	get_property(DISPLAYED GLOBAL PROPERTY ETH_${DEP}_DISPLAYED)
 	if (NOT DISPLAYED)
 		set_property(GLOBAL PROPERTY ETH_${DEP}_DISPLAYED TRUE)
+		if (NOT("${${DEP}_VERSION}" STREQUAL ""))
+			message(STATUS "${NAME} version: ${Boost_VERSION}")
```
Comment (ekpyron) [inline cmake/EthDependencies.cmake]: Shouldn't this be ``${${DEP}_VERSION}`` instead of ``Boost_VERSION`` here?

**Review by ekpyron — CHANGES_REQUESTED** [review]

**Review by ekpyron — APPROVED** [review]
Actually, do you think  we should provide some way of verifying the packages under https://github.com/ethereum/cpp-dependencies/releases/download/vs2017/?
I mean, if I understand you correctly, you just uploaded a local build of yours - if I'm pedantic, I don't know whether you intentionally uploaded something malicious or the system you built it on was compromised ;-).

But yeah - something that may be worth thinking about, but the PR here is fine.

**Comment by christianparpart** [comment]
@ekpyron you're not entirely wrong. But i believe there is no process for that (yet). Even the prior releases haven't been really created "automatically" by the CI (as it should, as of reading the build/CI scripts - it's "temporarily disabled" - since years). So we leave it as is for now, and at _some_ point in the future, we should globally overhaul and streamline the whole CI process :-)

**Pull request merged**